### PR TITLE
Add qcat red recipe

### PR DESCRIPTION
**Pull request by kjsanger**
Recipe is in a directory parallel to `recipes` because it requires conda-forge and bioconda

### CODE REVIEW

**Comment by dkj** [comment]
This doesn't show up when running `scripts/package_sort.py`. It that intended? Is that okay?

File: red-recipes/qcat/1.1.0-3-gbde56b5/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{% set version = "v1.1.0_3_gbde56b5" %}
```
Comment (dkj) [inline red-recipes/qcat/1.1.0-3-gbde56b5/meta.yaml:1]:
```suggestion
{% set version = "1.1.0_3_gbde56b5" %}
```
drop `v` to match versions in our other recipes.

Comment (kjsanger) [reply red-recipes/qcat/1.1.0-3-gbde56b5/meta.yaml:1]: I think that's fair. I've already had to mutate it from the original git version to eliminate the hyphens.

**Comment by kjsanger** [comment]
The `package_sort.py` script does have some kind of bug. It correctly spots that there are dependencies it doesn't know about, but it should print the name and version.

```
./scripts/package_sort.py --debug red-recipes/
DEBUG:root:recipe_file: red-recipes/qcat/1.1.0-3-gbde56b5/meta.yaml
INFO:root:Working on red-recipes/qcat/1.1.0-3-gbde56b5/meta.yaml
DEBUG:root:package name: qcat 1.1.0_3_gbde56b5
DEBUG:root:Requirements for qcat are: ['python', 'biopython', 'parasail-python', 'six', 'pyyaml', 'pip']
DEBUG:root:qcat version: 1.1.0_3_gbde56b5 requires python version: None
DEBUG:root:qcat version: 1.1.0_3_gbde56b5 requires biopython version: None
DEBUG:root:qcat version: 1.1.0_3_gbde56b5 requires parasail-python version: None
DEBUG:root:qcat version: 1.1.0_3_gbde56b5 requires six version: None
DEBUG:root:qcat version: 1.1.0_3_gbde56b5 requires pyyaml version: None
DEBUG:root:qcat version: 1.1.0_3_gbde56b5 requires pip version: None
DEBUG:root:('qcat', <LegacyVersion('1.1.0_3_gbde56b5')>) requires [('python', None), ('biopython', None), ('parasail-python', None), ('six', None), ('pyyaml', None), ('pip', None)]
WARNING:root:Can't find required package python
WARNING:root:Can't find required package biopython
WARNING:root:Can't find required package parasail-python
WARNING:root:Can't find required package six
WARNING:root:Can't find required package pyyaml
WARNING:root:Can't find required package pip
```

**Comment by kjsanger** [comment]
I've worked out why this recipe is not in the dependency graph. Normally, packages with no dependencies are added to the root of the dependency graph. However in this case it has several dependecies, but none have recipes (which the script punts on the grounds that the user knows what they are doing), so it never gets added to the graph. I'll fix this and add to the PR

**Comment by kjsanger** [comment]
```
$ ./scripts/package_sort.py red-recipes/
qcat 1.1.0_3_gbde56b5 red-recipes/qcat/1.1.0-3-gbde56b5
```

**Comment by dkj** [comment]
Note 0441414, is similar to https://github.com/wtsi-npg/npg_conda/pull/148/commits/71f21081895318b064146eb5585b895a40c5a9b9 + https://github.com/wtsi-npg/npg_conda/pull/148/commits/983288b1ca7487b74454543d19b1461a7080bec8 but I drop the URL requirement so we can just use `bioconda` or `conda-forge` strings.

**Comment by kjsanger** [comment]
Ah, that's not good. Should be straightforward to fix.

**Comment by kjsanger** [comment]
I changed the default to an empty iterable, which has fixed this.

**Pull request merged**